tests/Tokenizer/TokensTest.php

```diff
@@ -1910,6 +1910,34 @@ public function testSettingSizeInTryCatchBlockDoesNotChangeSize(): void
         self::assertSame($size, $tokens->getSize());
     }
 
+    public function testSettingSizeCachePruning(): void
+    {
+        $tokens = Tokens::fromCode('<?php $a = function () {};');
+
+        self::assertSame(11, $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, 10));
+        self::assertSame(10, $tokens->findBlockStart(Tokens::BLOCK_TYPE_CURLY_BRACE, 11));
+        self::assertTrue($tokens->isTokenKindFound(T_FUNCTION));
+        self::assertTrue($tokens->isTokenKindFound('{'));
+        self::assertTrue($tokens->isTokenKindFound('}'));
+        self::assertTrue($tokens->isTokenKindFound(';'));
+        self::assertFalse($tokens->isTokenKindFound(T_CLASS));
+
+        \Closure::bind(static function () use ($tokens): void {
+            $tokens->updateSize(\count($tokens) - 2);
```

Comment:
don't call internals. have a proper test for public interface to cover what you want to cover here